Makefile

```diff
@@ -29,19 +24,17 @@ inplace:
 	$(PYTHON) setup.py build_ext -i
 
 test-code: in
-	$(NOSETESTS) -s -v sklearn
+	$(PYTEST) --showlocals -v sklearn
```

Comment:
this doesn't show warnings, right?

Reply:
Not sure what you mean:
* pytest logs by default show a lot less warnings that nosetests. A good thing in my opinion
* for now, we disable pytest warnings in setup.cfg. This is mostly to get rid of the warnings about pytest not supporting tests using yield in pytest 4.0. This will be addressed in further PRs.

Reply:
I wasn't able to turn on all the warnings we get with nosetests, and I think that's a bad thing. For example I couldn't get deprecation warnings (ours or numpy) to show. But I'm new to pytest

Reply:
From https://docs.pytest.org/en/latest/warnings.html:
> Starting from version 3.1, pytest now automatically catches warnings during test execution and displays them at the end of the session:

and

> Pytest by default catches all warnings except for DeprecationWarning and PendingDeprecationWarning.

What I was not fully aware of this but it seems that `--disable-pytest-warnings` completely disables the warning summary. It was originally added to remove warnings about yield-based tests.

Reading more carefully https://docs.pytest.org/en/latest/warnings.html, it looks like you can use `filterwarnings` in setup.cfg to filter warnings a bit more carefully (only filter out yield-based tests deprecation and show other DeprecationWarnings). This is what we should probably do. Do you mind if I do that in a separate PR?

Reply:
Looking closely at it, removing `--disable-pytest-warnings` from setup.cfg you get 10k lines of warnings in the warnings summary when running `pytest sklearn`. I am not really in favor of doing that by default:
* it makes the pytest output a lot more noisy 
* you will not be able to see the Travis log directly inside Travis (you need to download the full log because it goes over the size limit). This is what is happening at the moment for example compare this [pytest log](https://travis-ci.org/scikit-learn/scikit-learn/jobs/302837490) to its [nose equivalent log](https://travis-ci.org/scikit-learn/scikit-learn/jobs/302837489). In the nose log you get `This log is too long to be displayed. Please reduce the verbosity of your build or download the raw log.`.  
* people interested to chase down warnings can run locally with the `--disable-pytest-warnings` commented out in their setup.cfg.

Side comment: actually we get less than 100 "yield tests are deprecated, and scheduled to be removed in pytest 4.0" warnings. So the original motivation for `--disable-pytest-warnings` does no longer exist.

Reply:
I think we should make sure we don't use warnings instead of ignoring them. Many of them are important. Hiding them hides the problem.

Reply:
Looking at the output of nose, I see that many of those are pretty recently introduced and probably point to real issues....